### PR TITLE
Update AdminWebClient and WebClient Snow Plow calls to new event mappings

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#16088](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/16088)

## Description

Updated SnowPlow mappings to AdminWebClient and WebClient;

- See [wiki](https://github.com/bcgov/healthgateway/wiki/snow-plow-event-mapping) for changes
- Deleted Snowplow Utility from AdminWebClient and replaced with ITrackingServiice
- Deleted Snowplow and EventTracker Utility from WebClient and replaced with ITrackingService

Note: EventData.destination is typed to Destination and Dataset enums.  Most scenarios use Destination but one scenario in Home Quicklinks uses Dataset

Example of what new calls look like:

Screenshot 2023-11-08 at 4 25 30 PM

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

None

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/components/public/vaccine-card/VaccineCardView.vue [hunk 1]
```diff
@@ -184,10 +188,11 @@ function downloadImage(): void {
...
-
-        SnowPlow.trackEvent({
-            action: "save_qr",
-            text: "vaxcard",
+        trackingService.trackEvent({
+            action: Action.Download,
+            text: Text.Document,
+            type: Type.PublicCovid19ProofOfVaccination,
+            format: Format.Image,
         });
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/public/vaccine-card/VaccineCardView.vue:196]:
```suggestion
        trackingService.trackEvent({
            action: Action.Download,
            text: Text.Document,
            type: Type.PublicCovid19ProofOfVaccination,
            format: Format.Image,
            actor: Actor.User,
        });
```

File: Apps/WebClient/src/ClientApp/src/stores/specialAuthorityRequest.ts [hunk 2]
```diff
@@ -135,10 +141,11 @@ export const useSpecialAuthorityRequestStore = defineStore(
...
                     }
-                    EventTracker.loadData(
-                        EntryType.SpecialAuthorityRequest,
-                        result.resourcePayload.length
-                    );
+                    trackingService.trackEvent({
+                        action: Action.Load,
+                        text: Text.Data,
+                        dataset: Dataset.SpecialAuthorityRequests,
+                    });
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/stores/specialAuthorityRequest.ts:148]: `EventTracker.loadData()` would only track a load event if at least 1 record was returned for the dataset. That logic may need to be included here to preserve the same behaviour.

(Applies to all dataset stores.)

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/src/components/private/home/HomeView.vue [hunk 3]
```diff
@@ -313,8 +340,12 @@ function handleClickQuickLink(index: number): void {
         .filter((d): d is EntryTypeDetails => d !== undefined);
 
     if (detailsCollection.length === 1) {
-        const linkType = detailsCollection[0].eventName;
-        trackClickLink(linkType);
+        trackingService.trackEvent({
+            action: Action.Visit,
+            text: Text.InternalLink,
+            destination: EventDataUtility.getDataset(detailsCollection[0].type),
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/components/private/home/HomeView.vue:346]: Changed.

File: Apps/WebClient/src/ClientApp/src/components/private/reports/ReportsComponent.vue [hunk 4]
```diff
@@ -239,17 +243,14 @@ function downloadReport(): void {
 }
 
 function trackDownload(): void {
-    let reportEventName =
-        entryTypeMap.get(selectedEntryType.value)?.reportEventName ?? "";
-
     const formatTypeName = ReportFormatType[reportFormatType.value];
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/components/private/reports/ReportsComponent.vue:246]: Changed.

File: Apps/WebClient/src/ClientApp/src/stores/specialAuthorityRequest.ts (hunk 2 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/stores/specialAuthorityRequest.ts]: Changed.

**Review by MikeLyttle — APPROVED** [review]

**Review by BronzBierd — DISMISSED** [review]
It looks great,

A few things I'm not a great fan of but no need to hold back on this one:

1. Putting all the types under `plugins/extensions`, I feel this should live its own dedicated tracking space now `plugins/tracking`
2. To remove the rather verbose requirement for each click. Custom directives could help capture some of the repetitiveness in the business logic and put the tracking consistently in a single location. (https://vuejs.org/guide/reusability/custom-directives.html) `hg-tracking:click` could attach and detach a listener for the targeted component. The argument could change depending on the type of listener required, for now only click would really help.

**Review by MikeLyttle — APPROVED** [review]

**Review by BronzBierd — APPROVED** [review]

**Pull request merged**